documentation/api/addAssertion.md

```diff
@@ -10,82 +10,208 @@ expect.addAssertion(pattern, handler);
-section for more information about the type system in Unexpected.
-
-A word in square brackets represents a flag that can either be
-there or not. If the flag is present `expect.flags[flag]` will contain
-the value `true`. In this case `not` is a flag. When a flag it present
-in a nested `expect` it will be inserted if the flag is present;
-otherwise it will be removed. Text that is in parentheses with
-vertical bars between them are treated as alternative texts that can
-be used. In this case you can write _ordered_ as an alternative to
-_sorted_.
+describing the pattern(s) the assertion should match. The pattern takes the
+following structure:
+
+`<the-subject> an assertion string <the-value>`
+
+The words in angle brackets define what types the assertion applies to. These
+can be any of the internally-defined types or new types added via
+[expect.addType](../addType). In this example, the subject to `to have item`
+must be an array, while the value may be of any type.
+
+If mismatching types are used, Unexpected throws an error with a helpful
+suggestion:
+
+```js
+expect('abcd', 'to have item', 'a');
+```
+
+```output
+expect('abcd', 'to have item', 'a');
+The assertion does not have a matching signature for:
+    <string> to have item <string>
+  did you mean:
+    <array> to have item <any>
+```
+
+If no subject type is specified, the assertion will be defined for the
+type `any` and would be applicable to any type.
 
 An assertion can only have a single assertion string, which must be
-provided after the subject type. This means that you cannot have
-more words, flags, and alternations after the first type.
+provided after the subject type. This means it's not possible to add assertions
+such as `<number> to be between <number> and <number>`.
+
+## Alternations
+
+Different versions of the same assertion, or different assertions that share the
+same handler function, can be added using an array:
+
+<!-- unexpected-markdown evaluate:false -->
+<!-- eslint-skip -->
+
+```js
+expect.addAssertion(
+  ['<array> to have item <any>', '<array> to have value <any>'],
+  /* handler */
+);
+```
+
+However, array patterns are ideal for assertions that have different types for
+the subject or value; for instance, if the second assertion was defined as
+`<array> to have value <number>`. When the difference is only in the assertion
+string, an alternation is more handy:
+
+<!-- unexpected-markdown evaluate:false -->
+<!-- eslint-skip -->
+
+```js
+expect.addAssertion('<array> to have (item|value) <any>', /* handler */);
+```
+
+Alternations are made available to the handler function as an
+`expect.alternations` array, which contains the word used when the assertion is invoked.
+
+## Flags
+
+Flags allow defining different versions of an assertion, where some words can
+be included when the assertion is invoked to yield different behaviour:
+
+<!-- unexpected-markdown evaluate:false -->
```

Comment:
Couldn't we get some of these to run? It's so comforting to have the examples tested :innocent:

Is it because you're defining the same assertion in multiple code blocks, and that would error out if they were executed? Then there's the handy `<!-- unexpected-markdown freshExpect:true -->` annotation that you can put before a code block and get a freshly cloned, new car-smelling `expect` :)

Reply:
It was precisely for that reason. Thank you for the fresh, new car-smelling syntax :)